services/sourceforge/sourceforge-last-commit.service.js

```diff
@@ -0,0 +1,63 @@
+import Joi from 'joi'
+import { BaseJsonService } from '../index.js'
+import { formatDate } from '../text-formatters.js'
+import { age as ageColor } from '../color-formatters.js'
+
+const schema = Joi.object({
+  timeline: Joi.array()
+    .items(
+      Joi.object({
+        published: Joi.number().required(),
+        verb: Joi.string().required(),
+      })
+    )
+    .required(),
+})
+
+export default class SourceforgeLastCommit extends BaseJsonService {
+  static category = 'activity'
+
+  static route = {
+    base: 'sourceforge/last-commit',
+    pattern: ':project',
+  }
+
+  static examples = [
+    {
+      title: 'SourceForge last commit',
+      namedParams: {
+        project: 'guitarix',
+      },
+      staticPreview: this.render({
+        commitDate: 1653556285,
+      }),
+    },
+  ]
+
+  static defaultBadgeData = { label: 'last commit' }
+
+  static render({ commitDate }) {
+    return {
+      message: formatDate(new Date(commitDate)),
+      color: ageColor(new Date(commitDate)),
+    }
+  }
+
+  async fetch({ project }) {
+    return this._requestJson({
+      url: `https://sourceforge.net/rest/p/${project}/activity`,
+      schema,
+      errorMessages: {
+        404: 'project not found',
+      },
+    })
+  }
+
+  async handle({ project }) {
+    const body = await this.fetch({ project })
+    return this.constructor.render({
+      commitDate: body.timeline.find(objet => objet.verb !== 'posted')
```

Comment:
What are the possible values of `verb` here? Do you have a link to the docs?

Reply:
I didn't find any documentation for this part of the api because I didn't realize that `/activity ` was outdated.
I corrected the two files that were calling it with `/git/commits` which is obviously much better documented because it is up to date.